x-pack/plugins/elastic_assistant/server/routes/chat/chat_complete_route.ts

```diff
@@ -151,20 +149,10 @@ export const chatCompleteRoute = (
           });
 
           let updatedConversation: ConversationResponse | undefined | null;
-          // Fetch any tools registered by the request's originating plugin
-          const pluginName = getPluginNameFromRequest({
-            request,
-            defaultPluginName: DEFAULT_PLUGIN_NAME,
-            logger,
-          });
-          const enableKnowledgeBaseByDefault =
-            ctx.elasticAssistant.getRegisteredFeatures(pluginName).assistantKnowledgeBaseByDefault;
+
           // TODO: remove non-graph persistance when KB will be enabled by default
-          if (
-            (!enableKnowledgeBaseByDefault || (enableKnowledgeBaseByDefault && !conversationId)) &&
-            request.body.persist &&
-            conversationsDataClient
-          ) {
+          // TODO: @yuliia, is this still needed for the `!conversationId` case that was for when enableKnowledgeBaseByDefault:true?
+          if (!conversationId && request.body.persist && conversationsDataClient) {
```

Comment:
@YulNaumenko, had a comment here for you. I believe this is still needed (as you [have it here](https://github.com/elastic/kibana/pull/188172/files#diff-2428b358355b5488d015da98734f504c1ad0f636b5399827bda12085b837a4d4R164-R165) as well), but wanted to double check with the other above comment about removal.

Reply:
Yes, it still needed, I'm thinking about the best approach how we can inject persistence flow - as graph/sub-graph/callbacks.